src/rust/src/incremental_merkle_tree.rs

```diff
@@ -29,25 +33,54 @@ pub fn write_checkpoint_v1<W: Write>(mut writer: W, checkpoint: &Checkpoint) ->
     Ok(())
 }
 
-pub fn read_checkpoint_v1<R: Read>(mut reader: R) -> io::Result<Checkpoint> {
+pub fn read_checkpoint_v1<R: Read>(
+    mut reader: R,
+    save_positions: &mut BTreeMap<usize, Position>,
+) -> io::Result<Checkpoint> {
+    let bridges_len = read_leu64_usize(&mut reader)?;
+    let is_witnessed = reader.read_u8()? == 1;
+    let forgotten = Vector::read_collected(&mut reader, |mut r| {
+        Ok((read_position(&mut r)?, read_leu64_usize(&mut r)?))
+    })?;
+
+    // v1 checkpoints lacked information about the witnesses that were made during their
+    // span of blocks, so we assume that a witness was made in the first checkpoint
+    // with a corresponding bridge index.
```

Comment:
In a future PR we should clarify this comment. This assumption stems from the fact that we witness immediately after a checkpoint in practice, rather than checkpointing more than one time before witnessing. Not blocking.